app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/AddCoInsuredBottomSheetContent.kt

```diff
@@ -58,82 +59,92 @@ import kotlinx.datetime.toLocalDateTime
+  onAddNewCoInsured: () -> Unit,
+  onCoInsuredSelected: (CoInsured) -> Unit,
 ) {
   Column(
-    horizontalAlignment = Alignment.CenterHorizontally,
     modifier = Modifier.padding(horizontal = 16.dp),
   ) {
     Spacer(Modifier.height(16.dp))
-    Text(stringResource(id = R.string.CONTRACT_ADD_COINSURED))
+    Text(
+      text = stringResource(id = R.string.CONTRACT_ADD_COINSURED),
+      textAlign = TextAlign.Center,
+      modifier = Modifier.fillMaxWidth(),
+    )
     Spacer(Modifier.height(24.dp))
-    AnimatedVisibility(visible = showManualInput) {
-      ManualInputFields(
-        birthDate = birthDate,
-        errorMessage = errorMessage,
-        onBirthDateChanged = onBirthDateChanged,
-        onFirstNameChanged = onFirstNameChanged,
-        onLastNameChanged = onLastNameChanged,
-      )
-    }
-    AnimatedVisibility(visible = !showManualInput) {
-      FetchFromSsnFields(
-        firstName = firstName,
-        lastName = lastName,
-        errorMessage = errorMessage,
-        onSsnChanged = onSsnChanged,
-        onContinue = onContinue,
+    if (bottomSheetState.canPickExistingCoInsured() && bottomSheetState.selectableCoInsured != null) {
+      SelectableCoInsuredList(
+        bottomSheetState.selectableCoInsured,
+        bottomSheetState.selectedCoInsured,
+        bottomSheetState.errorMessage,
+        onAddNewCoInsured,
+        onCoInsuredSelected,
       )
-    }
-    Spacer(Modifier.height(4.dp))
-    HedvigCard(
-      Modifier
-        .fillMaxWidth()
-        .clickable {
-          onManualInputSwitchChanged(!showManualInput)
-        },
-    ) {
-      HorizontalItemsWithMaximumSpaceTaken(
-        modifier = Modifier.padding(horizontal = 16.dp, vertical = 4.dp),
-        startSlot = {
-          Row(
-            verticalAlignment = Alignment.CenterVertically,
-          ) {
-            Text(
-              text = stringResource(id = R.string.CONTRACT_ADD_COINSURED_NO_SSN),
-              color = MaterialTheme.colorScheme.onSurfaceVariant,
+    } else {
+      AnimatedVisibility(visible = bottomSheetState.showManualInput) {
+        ManualInputFields(
+          birthDate = bottomSheetState.birthDate,
+          errorMessage = bottomSheetState.errorMessage,
+          onBirthDateChanged = onBirthDateChanged,
+          onFirstNameChanged = onFirstNameChanged,
+          onLastNameChanged = onLastNameChanged,
+        )
+      }
+      AnimatedVisibility(visible = !bottomSheetState.showManualInput) {
+        FetchFromSsnFields(
+          firstName = bottomSheetState.firstName,
+          lastName = bottomSheetState.lastName,
+          errorMessage = bottomSheetState.errorMessage,
+          onSsnChanged = onSsnChanged,
+          onContinue = onContinue,
+        )
+      }
+      Spacer(Modifier.height(4.dp))
+
+      HedvigCard(
+        Modifier
+          .fillMaxWidth()
+          .clickable {
+            onManualInputSwitchChanged(!bottomSheetState.showManualInput)
```

Comment:
Also a case where the HedvigCard whcih takes an `onClick` could be used instead.